src/pytest_subtests/plugin.py

```diff
@@ -275,7 +272,9 @@ def __exit__(
                 node=self.request.node, call=call_info, report=sub_report
             )
 
-        return True
+        if exc_val is not None:
+            if self.request.session.shouldfail:
+                pytest.exit(reason=f"subtest failed")
```

Comment:
This doesn't seem right, but seems to leave the parent test off the test result entirely.